File: bestagjofin.cpp
#include <iostream>

using namespace std;

int main()
{
    int guests{ 0 }, topNum{ 0 }, tempNum{ 0 };

    string topName, tempName;

    cin >> guests;

    for (int i = 0; i < guests; ++i)
    {
        cin >> tempName;
        cin >> tempNum;

        if (topNum < tempNum)
        {
            topName = tempName;
            topNum = tempNum;
        }
    }

    cout << topName;

    return 0;
}

File: bestagjofin.py
guests = int(input())

topNum = 0
topName = ""

for x in range(guests):
    input_str = input()
    parts = input_str.split()
    tempName = parts[0]
    tempNum = int(parts[1])

    if topNum < tempNum:
        topNum = tempNum
        topName = tempName

print(topName)
